UnityProject/Assets/Scripts/Players/PlayerHealthBar.cs

```diff
@@ -1,158 +1,161 @@
-
-		// Parent health bar positioning.
-		healthBar.transform.localRotation = Quaternion.Euler(
-			HealthBarRotationX, 
-			HealthBarRotationY, 
-			HealthBarRotationZ);
-		healthBar.transform.localPosition = new Vector3(
-			HealthBarPositionX, 
-			HealthBarPositionY, 
-			HealthBarPositionZ);		
-	}
-	
-	// Change how the health bar looks. Called from player controller.
-	public void SetHealthPoints(float hp)
-	{
-		hp *= (MaxHP - MinHP) / (MaxHPFromBackend - MinHPFromBackend);
-
-		// Bound hp from 0 to 10.
-		float actualHP = hp > MaxHP ? MaxHP : (hp < MinHP ? MinHP : hp);
-
-		// This is the formula so that it looks nice.
-		float greenBarX = GetGreenBarX(actualHP);
-		greenBar.transform.localPosition = new Vector3(
-			greenBarX, 
-			BarVerticalPosition, 
-			0.0f);
-		greenBar.transform.localScale = new Vector3(
-			actualHP, 
-			BarMaxScaleY, 
-			1.0f);
-	}
-
-	// Method that calculates how much we need to move the green bar when its
-	// scale changes.
-	private float GetGreenBarX(float hp)
-	{
-		return GreenBarPositionZeroHP + (hp * GreenBarIncrementOneHP);
-	}
+
+    /* Creates the health bar sprite attached to each avatar and provides an easy
+     * interface to modify it. When the health points change, it suffices to call
+     * SetHealthPoints() and this class will do the rest.
+     */
+
+    public class PlayerHealthBar : MonoBehaviour
+    {
+        // Texture constants.
+        private const int TextureWidth = 100;
+        private const int TextureHeight = 25;
+
+        // Sprite initialisation constants.
+        private const float SpriteRectWidth = 10.0f;
+        private const float SpriteRectHeight = 1.0f;
+        private const float SpritePivotCoordX = 0.5f;
+        private const float SpritePivotCoordY = 0.5f;
+        private const float SpritePixelsPerUnit = 100.0f;
+
+        // General bar constants.
+        private const float BarVerticalPosition = -0.3f;
+        private const float BarMaxScaleX = 10.0f;
+        private const float BarMaxScaleY = 10.0f;
+
+        // Health bar game object positioning constants.
+        private const float HealthBarRotationX = 0.0f;
+        private const float HealthBarRotationY = 0.0f;
+        private const float HealthBarRotationZ = 0.0f;
+        private const float HealthBarPositionX = 0.0f;
+        private const float HealthBarPositionY = 0.0f;
+        private const float HealthBarPositionZ = -0.45f;
+
+        // Constants to handle changes in the health poitns.
+        private const float MinHP = 0.0f;
+        private const float MaxHP = 10.0f;
+        private const float MinHPFromBackend = 0.0f;
+        private const float MaxHPFromBackend = 10.0f;
+        private const float GreenBarPositionZeroHP = -0.5f;
+        private const float GreenBarIncrementOneHP = 0.05f;
+
+        // Health bar game objects.
+        private GameObject healthBar;
+        private GameObject redBar;
+        private GameObject greenBar;
```

Comment:
Matter of preference really, but with this amount of constants we might as well create a separate package for them, and a class for each type?